app/classifier/tasks/shortcut-editor.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+import Shortcut from './shortcut';
+
+export default class ShortcutEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addAnswer = this.addAnswer.bind(this);
+    this.toggleShortcut = this.toggleShortcut.bind(this);
+  }
+
+  toggleShortcut(e) {
+    let nextTaskID;
+
+    if (e.target.checked) {
+      const taskCount = Object.keys(this.props.workflow.tasks).length;
+
+      let taskIDNumber = -1;
+
+      while (!((nextTaskID != null) && !(nextTaskID in this.props.workflow.tasks))) {
+        taskIDNumber += 1;
+        nextTaskID = `T${taskCount + taskIDNumber}`;
+      }
+
+      const changes = {};
+      changes[`tasks.${nextTaskID}`] = Shortcut.getDefaultTask(this.props.task.question);
+
+      this.props.task.unlinkedTask = nextTaskID;
+      this.props.workflow.update(changes);
+      this.addAnswer();
+    } else {
+      delete this.props.workflow.tasks[this.props.task.unlinkedTask];
+      delete this.props.task.unlinkedTask;
+      this.props.workflow.update('tasks');
+    }
+  }
+
+  addAnswer() {
+    this.props.workflow.tasks[this.props.task.unlinkedTask].answers.push({ label: 'Nothing Here' });
+    this.props.workflow.update('tasks');
+  }
+
+  removeChoice(index) {
+    if (this.props.task) {
+      const answers = this.props.workflow.tasks[this.props.task.unlinkedTask].answers;
+      answers.splice(index, 1);
+      if (!answers.length) {
```

Comment:
I'd prefer an explicit comparison here rather than a truthy value, `if (answers.length === 0)` should work.